schema/chat_room.proto

```diff
@@ -0,0 +1,343 @@
+}
+
+message GetRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+}
+
+message UpdateRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  Room room = 2;
+}
+
+message DeleteRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+}
+
+message ListRoomsRequest {
+  int32 page_size = 1;
+
+  string page_token = 2;
+}
+
+message ListRoomsResponse {
+  repeated Room rooms = 1;
+
+  string next_page_token = 2;
+}
+
+message StreamRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+}
+
+message SearchAllRequest {
+  string query = 1 [(google.api.required) = true];
+}
+
+message SearchAllMetadata {
+  google.rpc.RetryInfo retry_info = 1;
+}
+
+message SearchAllResponse {
+  repeated Message messages = 1;
+}
+
+message Message {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room"];
+
+  string name = 2 [(google.api.resource).path = "messages/*"];
+
+  // This will eventually become a User message that will cover add coverage for
+  // shared resources.
+  string author = 3;
+
+  string content = 4;
+
+  bytes image = 5;
+
+  google.protobuf.Timestamp create_time = 6;
+
+  google.protobuf.Timestamp update_time = 7;
+}
+
+message CreateMessageRequest {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  Message message = 2;
+}
+
+message GetMessageRequest {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  string name = 2 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Message",
+    (google.api.required) = true
+  ];
+}
+
+message UpdateMessageRequest {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  string name = 2 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Message",
+    (google.api.required) = true
+  ];
+
+  Message message = 3;
+}
+
+message DeleteMessageRequest {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  string name = 2 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Message",
+    (google.api.required) = true
+  ];
+}
+
+message ListMessagesRequest {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  string name = 2 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Message",
+    (google.api.required) = true
+  ];
+
+  int32 page_size = 3;
+
+  string page_token = 4;
+}
```

Comment:
Perhaps add `string filter` in a future version?

Reply:
Sure. I'll leave this out in this iteration.